config/v1/types_infrastructure.go

```diff
@@ -691,7 +691,32 @@ type BareMetalPlatformLoadBalancer struct {
 
 // BareMetalPlatformSpec holds the desired state of the BareMetal infrastructure provider.
 // This only includes fields that can be modified in the cluster.
-type BareMetalPlatformSpec struct{}
+type BareMetalPlatformSpec struct {
+	// apiServerInternalIPs are the IP addresses to contact the Kubernetes API
+	// server that can be used by components inside the cluster, like kubelets
+	// using the infrastructure rather than Kubernetes networking. These are the
+	// IPs for a self-hosted load balancer in front of the API servers. In dual
+	// stack clusters this list contains two IPs otherwise only one.
+	//
+	// +kubebuilder:validation:Format=ip
+	// +kubebuilder:validation:MaxItems=2
+	// +listType=set
+	APIServerInternalIPs []string `json:"apiServerInternalIPs"`
```

Comment:
This needs to have an optional tag right?
What happens when this list is empty?

Reply:
Added `optional`. As unsetting all VIPs is illegal right now but maybe in a future we allow that. Or not. In either case, I am not sure, should we explicitly write it in the description here or leave it up to our product docs?

Reply:
You should treat these godoc as documentation. So, I would expect something along the lines of explaining how these values get used.

```
// When omitted, values from the status.<field> will be used.
// Once set, cannot be removed.
```

We should add a validation using CEL for that second one.

And then on upgrade, it can be unset, and the CNO will just ignore it, but once it's set, the user can't unset it and the CNO will copy it over

Reply:
Fixed. Now the godoc should be self-explanatory and no additional documentation needed to understand this field